### ISSUE TITLE
[Feat] Use different output formats if needed, but do not misuse stderr

### ISSUE THREAD

**Issue by wenzeslaus** [OSGeo/grass#1129]
### Wrong output order in GUI

In GUI, asking for column names with `v.info -c` (in Console or in a module form/dialog) results in output where the output precedes the heading which is describing the output:

```
...
DOUBLE PRECISION|X_COORD
DOUBLE PRECISION|Y_COORD
CHARACTER|Z_VALUE
INTEGER|POLYGONID
DOUBLE PRECISION|SCALE
DOUBLE PRECISION|ANGLE
Displaying column types/names for database connection of layer <1>:
```

This applies to multiple modules. It is due to what goes to stdout and what to stderr in the module. GUI then gets these two outputs separately, not synced like in the command line.

### Misleading output in command line

However, in command line or when scripting, this leads to strange behavior too. Processing the output by piping leaves the `Displaying column types/names...` message there, although what you are looking at is something else:

```
GRASS > v.info -c gdc | grep DOUBLE | sed -e 's/.*|//g'
Displaying column types/names for database connection of layer <1>:
AREA
PERIMETER
X_COORD
Y_COORD
SCALE
ANGLE
```

### Seemingly wrong output when scripting

The misleading output turns into a unnecessary red flag (something you need to investigate to see if there was an issue) when you assign the result into a variable, e.g.:

```
GRASS > COLS=$(v.info -c gdc | grep DOUBLE | sed -e 's/.*|//g' | tr "\n" ",")
Displaying column types/names for database connection of layer <1>:
```

This, similarly to the previous one influences Python scripting as well as the interactive shell.

### stderr misuse

The cause of these issues is that stderr is misused for parts of the output. In the `v.info -c` case, `Displaying column types/names for database connection of layer <1>:` is clearly part of the output as it introduces what is being displayed and what data it is associated with. However, either is necessary part of the output and it should be therefore part of the output or it is not that important or it is even unwanted and in that case, it should not be printed/displayed at all.

### TLDR

In other words, do not misuse stderr for things which are part of the output, not error message - it is _standard error output_ after all.

### The current state

The motivation for the current state is to do more output with less configuration so that users don't have to care about setting a format when using it interactively and when scripting/using the API (and parsing the output further). However, I think it is doing more harm than good. Instead of settings of a format (e.g., `format=csv`, `format=pretty`, or `-s eval-like`, ...) which is pretty much standard for all programs producing outputs, the current solution is misusing stderr. This use of stdout-stderr distinction is unexpected (you need to fist understand that this is in fact the case), produces hard-to-interpret results (see above), and makes it at the end harder to use as an API (which seems to be the case in the GUI).

### Implementation

All modules which are producing output which needs extra information should either drop it, add it to parsable output, or provide two different outputs, one for interactive use and verbatim use in GUI and one for subsequent parsing in shell or Python.

This does need to be done for all modules, but more is better and it would increase consistency - consistency not only between the reworked modules, but even with the current modules which don't suffer from these issues. In some cases, it can be perhaps done in backwards compatible way (i.e., after the 8.0.0 release), especially if we won't require the pretty print to be the default. 

## Describe alternatives you've considered

There are several ways how to address this in terms of backwards compatibility, timing, outputs formats of individual modules, parameter handling. However, if we want the modules to behave as expected from any other command line program, we don't have any alternatives besides addressing this directly. 

## Additional context

See v.db.select for an example of a module where normal output is _not_ a combination of stdout and stderr and #1121 for how an option format can be implemented for "default" versus parsable output.
